kubelet/datadog_checks/kubelet/summary.py

```diff
@@ -0,0 +1,169 @@
+            pod_uid = pod.get('podRef', {}).get('uid')
+
+            if pod_namespace is None or pod_name is None or pod_uid is None:
+                self.log.warning("Kubelet returned incomplete results from '/stats/summary', missing data for POD: {}".format(pod))
+                continue
+
+            self._report_pod_stats(pod_namespace, pod_name, pod_uid, pod, instance_tags, main_stats_source)
+            self._report_container_stats(
+                pod_namespace, pod_name, pod.get('containers', []), pod_list_utils, instance_tags, main_stats_source
+            )
+
+    def _report_pod_stats(self, pod_namespace, pod_name, pod_uid, pod, instance_tags, main_stats_source):
+        pod_tags = tags_for_pod(pod_uid, tagger.ORCHESTRATOR)
+        if not pod_tags:
+            self.log.debug("Tags not found for pod: %s/%s - no metrics will be sent", pod_namespace, pod_name)
+            return
+        pod_tags += instance_tags
+
+        pod_ephemeral_usage = pod.get('ephemeral-storage')
+        if pod_ephemeral_usage:
+            used_bytes = pod_ephemeral_usage.get('usedBytes')
+            if used_bytes:
+                self.gauge(self.NAMESPACE + '.ephemeral_storage.usage', used_bytes, pod_tags)
+
+        # Metrics below should already be gathered by another mean (cadvisor endpoints)
+        if not main_stats_source:
+            return
+
+        network = pod.get('network')
+        if network:
+            rx_bytes = network.get('rxBytes')
+            if rx_bytes:
+                self.rate(self.NAMESPACE + '.network.rx_bytes', rx_bytes, pod_tags)
+            tx_bytes = network.get('txBytes')
+            if tx_bytes:
+                self.rate(self.NAMESPACE + '.network.tx_bytes', tx_bytes, pod_tags)
+
+    def _report_container_stats(
+        self, pod_namespace, pod_name, containers, pod_list_utils, instance_tags, main_stats_source
+    ):
+        # Metrics below should already be gathered by another mean (cadvisor endpoints)
+        if not main_stats_source:
+            return
+
+        for container in containers:
+            container_name = container.get('name')
+            if container_name is None:
+                self.log.warning(
+                    "Kubelet reported stats without container name for pod: %s/%s", pod_namespace, pod_name
+                )
+                continue
+
+            # No mistake, we need to give a tuple as parameter
+            container_id = pod_list_utils.get_cid_by_name_tuple((pod_namespace, pod_name, container_name))
+            if container_id is None:
+                self.log.debug(
+                    "Container id not found from /pods for container: %s/%s/%s - no metrics will be sent",
+                    pod_namespace,
+                    pod_name,
+                    container_name,
+                )
+                continue
+
+            # TODO: In `containers` we also have terminated init-containers, probably to be excluded?
```

Comment:
yes we should

Reply:
I checked on Linux and we don't consider them (but we do have them in the Docker check).